opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -11,3 +11,334 @@
+    for the exported metric are deferred.
+    """
+
+    def create_float_counter(self,
+                             name: str,
+                             description: str,
+                             unit: str,
+                             label_keys: LabelKeys,
+                             span_context: SpanContext = None
+                             ) -> 'CounterFloat':
+        """Creates a counter type metric that contains float values.
+
+        Args:
+            name: The name of the counter.
+            description: Human readable description of the metric.
+            unit: Unit of the metric values.
+            label_keys: list of keys for the labels with dynamic values.
+                Order of the list is important as the same order MUST be used
+                on recording when suppling values for these labels.
+            span_context: The `SpanContext` that identifies the `Span`
+                that the metric is associated with.
+
+        Returns: A new `CounterFloat`
+        """
+
+    def create_int_counter(self,
+                           name: str,
+                           description: str,
+                           unit: str,
+                           label_keys: LabelKeys,
+                           span_context: SpanContext = None
+                           ) -> 'CounterInt':
+        """Creates a counter type metric that contains int values.
+
+        Args:
+            name: The name of the counter.
+            description: Human readable description of the metric.
+            unit: Unit of the metric values.
+            label_keys: list of keys for the labels with dynamic values.
+                Order of the list is important as the same order MUST be used
+                on recording when suppling values for these labels.
+            span_context: The `SpanContext` that identifies the `Span`
+                that the metric is associated with.
+
+        Returns:
+            A new `CounterInt`
+        """
+
+    def create_float_gauge(self,
+                           name: str,
+                           description: str,
+                           unit: str,
+                           label_keys: LabelKeys,
+                           span_context: SpanContext = None
+                           ) -> 'GaugeFloat':
+        """Creates a gauge type metric that contains float values.
+
+        Args:
+            name: The name of the counter.
+            description: Human readable description of the metric.
+            unit: Unit of the metric values.
+            label_keys: list of keys for the labels with dynamic values.
+                Order of the list is important as the same order MUST be used
+                on recording when suppling values for these labels.
+            span_context: The `SpanContext` that identifies the `Span`
+                that the metric is associated with.
+
+        Returns:
+            A new `GaugeFloat`
```

Comment:
Minor personal opinion here, it might be better to use `create_float_guage` with `FloatGauge` rather than mixing "float gauge" and "gauge float". Up to you.

Reply:
Makes sense. I think I will leave this for the next PR, seeing as we are getting rid of TimeSeries and these naming changes should be done all at once for consistency.